### PR TITLE
Issue #987: Resolve PR->Issue number for review/merge phase events

### LINKED ISSUE #987: auto: review/merge フェーズのイベント集計に PR→Issue 番号解決を追加
## Background

batch session `18964-1783692542` (2026-07-10) の L3 retrospective で、review/merge phase のイベントが **PR 番号を `issue` フィールドに持つ**ことによる集計歪みを確認した:

- `get-auto-session-report.sh` の "Issues processed: 10" は実 Issue 7 件 + PR #978/#983/#985 の誤集計
- Sub-Issue Completion Timeline に Size "?/?" の PR 行が混入 (Issue と結合できないため)
- Tier 3 recovery イベントが `issue: 983` (PR 番号) に帰属し、実 Issue #970 との関連が集計上失われる
- `concurrent_commit_detected` も `issue: 978/983` で記録される

根本原因: `run-auto-sub.sh` が review/merge phase の wrapper を PR 番号で呼び出し、`run_phase_with_recovery` / wrapper 内の `emit_event` が `EMIT_ISSUE_NUMBER`=PR 番号でイベントを書くため。

これは同根クラスの第 3 の surface (第 1: #974 concurrent_commit 自己除外 — 修正済み、第 2: #984 recovery 記録 — open)。

検出元: `docs/sessions/18964-1783692542-2026-07-10/session.md` の Findings。

## Purpose

review/merge phase のイベントが実 Issue 番号で集計されるようにする。方式は **emit 時解決を採用** (根拠は Auto-Resolved Ambiguity Points 参照):
- **emit 時解決 (採用)**: run-auto-sub.sh が review/merge phase 呼び出し時に Issue 番号 (`$SUB_NUMBER`) をイベント文脈へ伝搬し、`issue` フィールドに実 Issue 番号を記録する (`pr` フィールドを別途追加してトレーサビリティ維持)
- 集計時解決 (不採用): `get-auto-session-report.sh` が PR 番号を `closes #N` 解決で実 Issue へマージする

#984 (recovery 記録の PR→Issue 解決、本 Issue 起票時点で open) とは独立に実装する。共通ヘルパ化は、いずれかが先に着地した時点でその実装を再利用する形の任意最適化とする (ブロッキング依存にはしない)。

### Pre-merge (auto-verified)
- [x]  review/merge イベントが emit 時点で実 Issue 番号を issue フィールドに記録する
- [x]  PR 番号の独立 Issue 混入を防ぐ regression テストが追加されている

### Post-merge
- [x] 次回 pr route を含む `/auto --batch` の L3 session report で、Issues processed が実 Issue 数と一致することを観察  

## Related Issues

- 検出元: batch session 18964-1783692542 の L3 retrospective
- 同根クラス: #974 (concurrent_commit 自己除外 — 修正済み)、#984 (recovery 記録 — open。共通ヘルパ化を検討)

## Auto-Resolved Ambiguity Points

- **実装方式: emit 時解決を採用** — `run-auto-sub.sh` が review/merge phase 呼び出し時に既知の `$SUB_NUMBER` (実 Issue 番号) を伝搬し、`run_phase_with_recovery` / `emit_event` の `issue` フィールドに実 Issue 番号を記録する。PR 番号は別途 `pr` フィールドとして保持しトレーサビリティを維持する。
  - 根拠: #974 の修正 (commit `fe753652`) が `_EXTRA_SELF_ISSUE="$SUB_NUMBER"` を `run_phase_with_recovery` に伝搬する同型のパターンを既に確立済み。呼び出し元 (`run-auto-sub.sh`) は review/merge phase 呼び出し時点で実 Issue 番号を既知であり、同じ経路で `issue` フィールドを解決するのが一貫性・低リスク。
  - 代替案 (不採用): 集計時解決 (`get-auto-session-report.sh` が PR→Issue マージ) — `orchestration-recoveries.md` や Spec Auto Retrospective への書き込み時点の誤りという根本原因を修正できないため不採用。
- **#984 との共通ヘルパ化: 独立実装とする** — #984 は本 Issue 起票時点で open。本 Issue は `emit_event` の `issue` フィールド解決という、より広いスコープ (phase_start/phase_complete/concurrent_commit_detected を含む全イベント種別) を持つため、#984 の着地を待たずに独立して実装する。いずれかが先に着地した場合、後続がその実装パターンを再利用して重複実装を避ける。
  - 根拠: 両 Issue とも「呼び出し元が実 Issue 番号を既知」という同一の根本原因クラスであり、AC テキストへの影響がないためブロッキング依存にはしない。
- **Post-merge observation AC への rubric 付与 (`/spec` Step 10 自動対応)** — 元の post-merge AC は `<!-- verify-type: observation event=auto-run -->` タグのみで、観測イベントと期待される出力構造の分離が prose 内に留まっていた (`modules/verify-classifier.md` の observation-tagged 条件チェック)。Option B (rubric verify command 付与) を採用し、`auto-run` イベント発火時の `/verify` 再評価が「Issues processed が実 Issue 数と一致し、PR 番号が独立 Issue として計上されていない」という具体的な構造基準を機械的に参照できるようにした。

### PR DESCRIPTION
**Pull request by saito**
## Summary

`run-auto-sub.sh` の review/merge phase 呼び出し (`run_phase_with_recovery "review" "$PR_NUMBER" ...` / `"merge" "$PR_NUMBER" ...`) は、`emit_event()` の `issue` フィールドに実 Issue 番号ではなく PR 番号を記録してしまっていた。これにより `get-auto-session-report.sh` の集計で PR 番号が独立した "Issue" として二重集計され、"Issues processed" 等のメトリクスが歪む問題があった (実例: batch session `18964-1783692542` で "Issues processed: 10" と誤集計)。

方式は **emit 時解決**: `run_phase_with_recovery()` が既に受け取っている `_EXTRA_SELF_ISSUE` (実 Issue 番号、#974 で導入済みの伝搬パターンを再利用) を使って `EMIT_ISSUE_NUMBER` を実 Issue 番号に解決し、PR 番号は新設の `EMIT_PR_NUMBER`/`pr` フィールドに別途記録してトレーサビリティを維持する。

- `scripts/emit-event.sh`: `emit_event()` に `EMIT_PR_NUMBER` が設定されている場合のみ `pr` フィールドを JSON に追加
- `scripts/run-auto-sub.sh`: `run_phase_with_recovery()` の `EMIT_ISSUE_NUMBER` 解決ロジックを `_EXTRA_SELF_ISSUE` 分岐に変更。`_maybe_emit_phase_complete()` の backfill JSON にも `pr` フィールドを追加
- `modules/event-emission.md`: `pr` フィールドと `EMIT_PR_NUMBER` の仕様を追記 (SSoT)
- `tests/emit-event.bats` / `tests/run-auto-sub.bats`: regression テストを追加

## Verification (pre-merge)
- review/merge phase のイベントが emit 時点で実 Issue 番号を issue フィールドに記録する仕組みが実装されていることを rubric で確認
- PR 番号の独立 Issue 混入を防ぐ regression テストが追加されていることを rubric で確認
- `bats tests/` フルスイート 1130 件すべて PASS (behavioral change 判定によりフルスイート実行)

## Verification (post-merge)
- 次回 pr route を含む `/auto --batch` の L3 session report で、Issues processed が実 Issue 数と一致することを観察

closes #987

### CODE REVIEW

File: tests/run-auto-sub.bats [hunk 1]
```diff
@@ -989,6 +989,31 @@ MOCK
     grep -q "concurrent_commit_detected.*commit_sha=ccc3333" "$BATS_TEST_TMPDIR/emit.log"
 }
 
+@test "review/merge phase events emit issue=<real Issue number> and pr=<PR number> (issue #987)" {
+    export AUTO_EVENTS_LOG="$BATS_TEST_TMPDIR/auto-events.jsonl"
+    export EMIT_ISSUE_NUMBER="42"
+
+    cat > "$MOCK_DIR/emit-event.sh" <<MOCK
+emit_event() {
+  echo "phase=\$EMIT_PHASE_NAME issue=\$EMIT_ISSUE_NUMBER pr=\${EMIT_PR_NUMBER:-<unset>} event=\$1" >> "$BATS_TEST_TMPDIR/emit.log"
```
Comment (saito) [inline tests/run-auto-sub.bats]: **[Edge Case/Robustness] MUST**: This regression test fails deterministically in a clean CI environment. `scripts/run-auto-sub.sh` runs under `set -euo pipefail` (line 7), and `emit_event "sub_start" "size=${SIZE}"` (scripts/run-auto-sub.sh:567) fires *before* `EMIT_PHASE_NAME` is ever exported (that export happens later, inside `run_phase_with_recovery()`). This mock references `$EMIT_PHASE_NAME` / `$EMIT_ISSUE_NUMBER` as bare (non-defaulted) variable expansions, so referencing the still-unset `$EMIT_PHASE_NAME` raises `EMIT_PHASE_NAME: unbound variable` under `set -u`, aborting the script before `emit.log` is ever created — confirmed as the actual CI failure (`gh run view` job 86551034931: `not ok 718 ... [ "$status" -eq 0 ]' failed`) and independently reproduced with `env -i`.

The failure is masked in ambient shells that already export `EMIT_PHASE_NAME`/`_EXTRA_SELF_ISSUE` (e.g. a nested `/auto` review-phase invocation), which is presumably why the PR description's local full-suite run reported all green.

Recommended fix — use nounset-safe defaults, matching the convention already used on the same line for `EMIT_PR_NUMBER`:
```bash
cat > "$MOCK_DIR/emit-event.sh" <<MOCK
emit_event() {
  echo "phase=\${EMIT_PHASE_NAME:-} issue=\${EMIT_ISSUE_NUMBER:-} pr=\${EMIT_PR_NUMBER:-<unset>} event=\$1" >> "$BATS_TEST_TMPDIR/emit.log"
}
_emit_comments_consumed() { :; }
MOCK
```

File: scripts/run-auto-sub.sh [hunk 2]
```diff
@@ -194,8 +194,10 @@ _maybe_emit_phase_complete() {
         '[.[] | select(.issue == $n)] | last // empty | .event // ""' 2>/dev/null || true)
   if [[ "${_last_event}" == "phase_start" ]]; then
     local _ts; _ts=$(date -u +%Y-%m-%dT%H:%M:%SZ)
+    local _pr_field=""
+    [[ -n "${EMIT_PR_NUMBER:-}" ]] && _pr_field=",\"pr\":${EMIT_PR_NUMBER}"
```
Comment (saito) [inline scripts/run-auto-sub.sh:198]: **[Edge Case/Robustness] SHOULD**: Spec Implementation Step 3 added this `pr` field to `_maybe_emit_phase_complete()`'s printf-based backfill JSON (the EXIT-trap path used on SIGTERM/watchdog kill), but neither `tests/run-auto-sub.bats` nor `tests/emit-event.bats` has a regression test exercising this backfill path with `EMIT_PR_NUMBER` set. This is new behavior (not a copy of an already-tested pattern), so a dedicated test would guard the SIGTERM/backfill path against future regressions.

File: modules/event-emission.md [hunk 3]
```diff
@@ -57,6 +57,28 @@ is written with `"backfilled": true`. This covers exit code 0 (clean exit) and e
 }
 ```
 
+### pr field (review/merge phase events)
+
+For review/merge phase events dispatched via `run-auto-sub.sh`'s `run_phase_with_recovery()`,
+the `issue` field always holds the real Issue number (not the PR number the phase was invoked
+with) — resolved from `_EXTRA_SELF_ISSUE` (see `_EMIT_PHASE_OWNED` pattern below). The PR number
```
Comment (saito) [inline modules/event-emission.md]: **[Documentation] CONSIDER**: This cross-reference ("see `_EMIT_PHASE_OWNED` pattern below") points to the wrong section — the `_EMIT_PHASE_OWNED` pattern (further below) documents the wrapper-side double-emit-avoidance pattern and never mentions `_EXTRA_SELF_ISSUE`; that variable/logic lives only in `run_phase_with_recovery()` in `run-auto-sub.sh`, described in the Wrapper Coverage Table's `run-auto-sub.sh` row instead. Consider pointing the cross-reference there.

**Review by saito — COMMENTED** [review]
## Acceptance Criteria Verification Results

| Condition | Result | Notes |
|-----------|--------|-------|
| review/merge phase イベントが emit 時点で issue フィールドに実 Issue 番号を記録し、PR 番号は別途 pr フィールドで保持 | PASS | `run_phase_with_recovery()` の `_EXTRA_SELF_ISSUE` 分岐、および同一関数スコープ内の phase_start/phase_complete/recovery(tier1-3)/concurrent_commit_detected の全 `emit_event` 呼び出しが解決済み値を継承することを確認 |
| PR 番号の独立 Issue 混入を防ぐ regression テストが追加されている | PASS | tests/emit-event.bats と tests/run-auto-sub.bats に該当テストが存在 (ただし後者は CI で FAIL — 下記 Code Review 参照) |
| (Post-merge) 次回 pr route を含む `/auto --batch` の L3 session report で Issues processed が実 Issue 数と一致 | POST-MERGE | `/verify` で event=auto-run 発火時に再評価 |

## CI Status

| Job | Status | Notes |
|-----|--------|-------|
| Run bats tests | FAILURE | `tests/run-auto-sub.bats:998` の新規テストが `set -u` 下で `$EMIT_PHASE_NAME` の無防備参照によりクラッシュ。下記 Code Review の MUST 指摘を参照 |
| DCO | SUCCESS | — |
| Validate skill syntax | SUCCESS | — |
| Forbidden Expressions check | SUCCESS | — |
| macOS shell compatibility | SUCCESS | — |

## Code Review

### General Comments (issues where path/line cannot be identified)

なし (全指摘は path/line 特定済みのためインラインコメントとして投稿)

### Issue Count Summary

- Posted as line comments: 3 (MUST: 1, SHOULD: 1, CONSIDER: 1)
- General Comments (in body): 0

---
MUST issues found. After fixing in Step 11 and posting response summary in Step 13, run `/merge 988`.

**Comment by saito** [comment]
## Review Response Summary

### Claude Review Response

| Aspect | Filename | Line | Issue | Response | Fix/Skip reason |
|--------|---------|------|-------|----------|----------------|
| Edge Case/Robustness (MUST) | tests/run-auto-sub.bats | 998 | 新規 regression テストのモックが `set -u` 下で `$EMIT_PHASE_NAME` を無防備参照し、`emit_event "sub_start"` が `EMIT_PHASE_NAME` 未設定時点で呼ばれるため CI (クリーン環境) で確実にクラッシュしていた | Resolved | `${EMIT_PHASE_NAME:-}` / `${EMIT_ISSUE_NUMBER:-}` の安全なデフォルト展開に修正。`env -i` のクリーン環境で PASS を確認 |
| Edge Case/Robustness (SHOULD) | scripts/run-auto-sub.sh | 198 | `_maybe_emit_phase_complete()` の backfill JSON への `pr` フィールド追加 (SIGTERM/watchdog kill 時のみ発火する EXIT trap パス) に対する直接的な regression テストがない | Skipped | EXIT trap 発火を単体テストで安全に再現するには追加の足場が必要で、リスク・工数に対し優先度が低いため見送り。SIGTERM/watchdog kill 時の稀な backfill レコードのみに影響する軽微な一貫性項目 |
| Documentation (CONSIDER) | modules/event-emission.md | 64 | `_EXTRA_SELF_ISSUE` の相互参照リンクが `_EMIT_PHASE_OWNED` パターン節 (無関係) を指していた | Resolved | 参照先を `run-auto-sub.sh` row in Wrapper Coverage Table に修正 |

### Lightweight Re-check Results

- Tests: `bats tests/*.bats` フルスイート 1130 件、クリーン環境 (`env -i`) で 0 failures (並列実行時に見られた 2 件の flaky failure は逐次実行で再現せず、本 PR の変更とは無関係と確認)
- `validate-skill-syntax.py skills/`: 0 error(s) (既存の unrelated warning 1件は本 PR のスコープ外)
- Fix diff check: PASS
- New issues: none
